### PR TITLE
Issue #204 -Setup CG SMTP Configuration 

### PR DESCRIPTION
**Pull request by Rohit90700**
Getting CG mail at the time of booking the table.

### CODE REVIEW

File: java/mtsj/core/src/main/java/com/devonfw/application/mtsj/mailservice/logic/impl/MailMock.java [hunk 1]
```diff
@@ -17,8 +16,9 @@
 @ConditionalOnProperty(prefix = "spring.mail", name = "enabled", havingValue = "false")
 public class MailMock implements Mail {
 
-  @Value("${spring.mail.username}")
-  private String from;
+  /*
+   * @Value("${spring.mail.username}") private String from;
```
Comment (hohwille) [inline java/mtsj/core/src/main/java/com/devonfw/application/mtsj/mailservice/logic/impl/MailMock.java]: Samehere. Actually looks quite redundant. Introduce common abstract class `AbstractMail` inherited from `MailImpl` and `MailMock`.
BTW: The name `Mail` is kind of odd and confusing. This is not a Mail/Email but a component that can send Emails. Hence it should be renamed. Actually this is a use-case in the logic-layer. It should therefore be `Mailmanagement` (with consistent package rename from `mailservice` or the interface should also be named `Mailservice`).

File: java/mtsj/core/src/main/resources/application.properties [hunk 2]
```diff
@@ -52,9 +52,14 @@ mythaistar.hourslimitcancellation=1
...
 #enabled set to false will use the MailMock service
-spring.mail.enabled=false
-spring.mail.host=smtp.gmail.com
-spring.mail.username=mythaistarrestaurant@gmail.com
-spring.mail.password=mythaistarrestaurant2501
-spring.mail.properties.mail.smtp.auth=true
-spring.mail.properties.mail.smtp.starttls.enable=true
+spring.mail.enabled=true
+#spring.mail.host=smtp.gmail.com
+spring.mail.host=de-nagios01.corp.capgemini.com
```
Comment (hohwille) [inline java/mtsj/core/src/main/resources/application.properties]: environment config should only go to `config/application.properties`. Please read the header of these properties files and guide:
https://github.com/devonfw/devon4j/blob/develop/documentation/guide-configuration.asciidoc#externalized-configuration

File: java/mtsj/core/src/main/resources/config/application.properties [hunk 3]
```diff
@@ -32,9 +32,13 @@ mythaistar.hourslimitcancellation=1
...
-spring.mail.properties.mail.smtp.auth=true
-spring.mail.properties.mail.smtp.starttls.enable=true
\ No newline at end of file
+spring.mail.enabled=true
+#spring.mail.host=smtp.gmail.com
+spring.mail.host=de-nagios01.corp.capgemini.com
+spring.mail.port=25
+#spring.mail.username=mythaistarrestaurant@gmail.com
+#spring.mail.password=mythaistarrestaurant2501
+#spring.mail.properties.mail.smtp.auth=true
```
Comment (hohwille) [inline java/mtsj/core/src/main/resources/config/application.properties]: Clean-Code: Remove outcommented stuff. It is all still in the git history and otherwise just spams the code.

File: angular/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,15976 @@
+{
+  "name": "mythaistar-restaurant",
```
Comment (hohwille) [inline angular/package-lock.json]: this commit does not seem related to the issue. Can we leave this out?

**Comment by hohwille** [comment]
Feedback from Yammer. We could consider this tool:
https://github.com/mailhog/MailHog
Can be easily integrated with docker...

**Comment by dario-rodriguez** [comment]
I used this tool: https://github.com/djfarrelly/MailDev
It works very well inside a docker container.
